plugins/catch_pokemon/__init__.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+import random
+import time
+
+from pokemongo_bot.human_behaviour import sleep
+from pokemongo_bot.event_manager import manager
+from pokemongo_bot import logger
+
+
+@manager.on("pokemon_found")
+def pokemon_found(bot, encounters=None):
+    # type: (PokemonGoBot, Optional[List[Encounter]]) -> None
+
+    if encounters is None or len(encounters) == 0:
+        return
+
+    def log(text, color="black"):
+        logger.log(text, color=color, prefix="Catch")
+
+    for pokemon_encounter in encounters:
+        encounter_id = pokemon_encounter['encounter_id']
+        spawn_point_id = pokemon_encounter['spawn_point_id']
+        player_latitude = pokemon_encounter['latitude']
+        player_longitude = pokemon_encounter['longitude']
+        bot.api_wrapper.encounter(encounter_id=encounter_id,
+                                  spawn_point_id=spawn_point_id,
+                                  player_latitude=player_latitude,
+                                  player_longitude=player_longitude)
+
+        response = bot.api_wrapper.call()
+
+        encounter_data = response["encounter"]
+        status = encounter_data.status
+        if status is 7:
+            # Pokemon bag is full, fire event and return for now
+            log("Pokemon bag is full; cannot catch.", color="red")
+            bot.fire("pokemon_bag_full")
+            return
```

Comment:
It's possible that this will actually not be the case in a couple of situations:

1. The Pokemon transfer plugin is not loaded or disabled by the user.

2. The user's CP/IV settings are set such that no Pokemon met the criteria to be transferred.

In either of those cases, the bot will be stuck in a loop of trying to catch, failing because the bag is full, transferring 0 Pokemon, and trying to catch again.